src/main/java/com/minecolonies/core/colony/buildings/modules/BuildingResourcesModule.java

```diff
@@ -353,13 +352,10 @@ public void checkOrRequestBucket(@Nullable final BuilderBucket requiredResources
                 continue;
             }
 
-            if (workerInv)
+            count += InventoryUtils.getItemCountInItemHandler(worker.getInventory(), stack -> ItemStackUtils.compareItemStacksIgnoreStackSize(stack, itemStack.getItemStack()));
```

Comment:
while on it we should use the Building specific Inventory Utils hasBuildingEnoughElseCount

Reply:
ncm

Reply:
Ray told me he was going to do further improvements himself.